exonum/src/events/codec.rs

```diff
@@ -64,3 +72,30 @@ impl Encoder for MessagesCodec {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod test {
+    use super::MessagesCodec;
+
+    use messages::{MessageBuffer, RawMessage};
+    use bytes::BytesMut;
+    use tokio_io::codec::Decoder;
+    #[test]
+    fn decode_message_valid_header_size() {
+        let data = vec![0u8, 0, 0, 0, 0, 0, 10, 0, 0, 0];
+        let mut bytes: BytesMut = data.as_slice().into();
+        let mut codec = MessagesCodec;
+        match codec.decode(&mut bytes) {
+            Ok(Some(ref r)) if r == &RawMessage::new(MessageBuffer::from_vec(data)) => {}
+            _ => panic!("Wrong input"),
+        };
+    }
+
+    #[test]
+    fn decode_message_small_size_in_header() {
```

Comment:
Either the test for `total_len > MAX_MESSAGE_LEN` is missing (then let's add it?) or it is somewhere else (then let's move this test over there as well?).


Also, this might be a nice use case for [proptest](https://github.com/AltSysrq/proptest) library.

Reply:
@matklad, the test for `total_len > MAX_MESSAGE_LEN` resides in `src/events/tests.rs` (`test_network_max_message_len`). Yea, I think, we should combine them in one file. This one seems to be better, because here we can check what error exactly we receive. While in `src/events/tests.rs` we can only check that there is some error, not knowing which one specifically.